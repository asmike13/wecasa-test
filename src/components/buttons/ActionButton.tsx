export interface IActionButtonProps<T> {
	setAction: (action: T) => void
	value: T
}

const ActionButton = ({ setAction, value }: IActionButtonProps<any>) => {
	return (
		<button data-testid="clickAction" onClick={setAction}>{value}</button>
	)
}

export default ActionButton
