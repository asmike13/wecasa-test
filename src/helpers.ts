export const timeFormat = (timeInMinutes: number) => {
	const hours = Math.floor(timeInMinutes / 60)
	const minutes = timeInMinutes % 60

	const displayHours = hours > 0 ? `${hours}H` : ''
	const displayMinutes = minutes > 0 ? `${minutes}min` : ''

	return `${displayHours} ${displayMinutes}`
}

export const priceFormat = (priceInCents: number) => {
	const price = Math.floor(priceInCents / 100)
	const cents = priceInCents % 100

	const displayPrice = price > 0 ? price : ''
	const displayCents = cents > 0 ? `.${cents}` : ''

	return `${displayPrice}${displayCents}â‚¬`
}
