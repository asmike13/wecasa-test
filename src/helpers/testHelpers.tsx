import { Provider } from 'react-redux'
import { store } from 'src/store'

export const StoreWrapper = (props: any) => {
	return (
		<Provider store={store}>
			{props.children}
		</Provider>
	)
}
