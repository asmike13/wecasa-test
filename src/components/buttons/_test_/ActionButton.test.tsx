import { fireEvent, render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer';
import { StoreWrapper } from 'src/helpers/testHelpers';
import ActionButton, { IActionButtonProps } from '../ActionButton';

export const props: IActionButtonProps<boolean> = {
	setAction: jest.fn(),
	value: false,
}

it('should render ActionButton as snapshot', async () => {
	const snapshot = renderer.create(
		<StoreWrapper>
			<ActionButton {...props} />
		</StoreWrapper>,
	)

	expect(snapshot).toMatchSnapshot()
})

it('should call setAction on click', async () => {
	render(
		<StoreWrapper>
			<ActionButton {...props} />
		</StoreWrapper>,
	)

	fireEvent.click(screen.getByTestId('clickAction'))
	expect(props.setAction).toHaveBeenCalledTimes(1)
})
