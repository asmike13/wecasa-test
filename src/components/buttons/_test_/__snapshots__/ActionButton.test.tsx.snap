// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ActionButton as snapshot 1`] = `
<button
  data-testid="clickAction"
  onClick={[MockFunction]}
/>
`;
